src/clr-angular/data/datagrid/render/row-renderer.ts

```diff
@@ -3,46 +3,27 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { AfterContentInit, ContentChildren, Directive, OnDestroy, QueryList } from '@angular/core';
+import { ContentChildren, Directive, OnDestroy, QueryList } from '@angular/core';
 import { Subscription } from 'rxjs';
 
-import { DatagridRenderStep } from '../enums/render-step.enum';
-
 import { DatagridCellRenderer } from './cell-renderer';
 import { DatagridRenderOrganizer } from './render-organizer';
+import { ColumnsService } from '../providers/columns.service';
 
 @Directive({ selector: 'clr-dg-row, clr-dg-row-detail' })
-export class DatagridRowRenderer implements AfterContentInit, OnDestroy {
-  constructor(private organizer: DatagridRenderOrganizer) {
-    this.subscriptions.push(
-      organizer.filterRenderSteps(DatagridRenderStep.ALIGN_COLUMNS).subscribe(() => this.setWidths())
-    );
-  }
-
+export class DatagridRowRenderer implements OnDestroy {
+  @ContentChildren(DatagridCellRenderer) cells: QueryList<DatagridCellRenderer>;
   private subscriptions: Subscription[] = [];
+
+  constructor(private organizer: DatagridRenderOrganizer, private columnsService: ColumnsService) {}
+
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
 
-  @ContentChildren(DatagridCellRenderer) cells: QueryList<DatagridCellRenderer>;
-
-  private setWidths() {
-    if (this.organizer.widths.length !== this.cells.length) {
-      return;
-    }
+  setupColumns() {
     this.cells.forEach((cell, index) => {
-      const width = this.organizer.widths[index];
-      cell.setWidth(width.strict, width.px);
-    });
-  }
-
-  ngAfterContentInit() {
-    this.cells.changes.subscribe(() => {
-      this.setWidths();
+      cell.columnState = this.columnsService.columns[index];
```

Comment:
Do we have a case where the cells of a row can change (I mean `this.cells.changes` fires) without the headers of the datagrid changing? If so, then you're missing a subscription here for cell ContentChildren changes.

Reply:
True, not a usual case I would guess but I'm sure it happens.